exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -307,3 +315,251 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
+			return errors.Wrapf(err, "failed to add watcher on infraMachine %q", infraMachine.GroupVersionKind())
+		}
+	}
+
+	return nil
+}
+
+// setInfraMachineOwnerRefs sets the ownerReferences on the each infraMachine to its associated MachinePool Machine.
+func (r *MachinePoolReconciler) setInfraMachineOwnerRefs(ctx context.Context, mp *expv1.MachinePool, infraMachines []unstructured.Unstructured) error {
+	machineList := &clusterv1.MachineList{}
+	labels := map[string]string{
+		clusterv1.MachinePoolNameLabel: mp.Name,
+		clusterv1.ClusterNameLabel:     mp.Spec.ClusterName,
+	}
+
+	if err := r.Client.List(ctx, machineList, client.InNamespace(mp.Namespace), client.MatchingLabels(labels)); err != nil {
+		return err
+	}
+
+	infraMachineNameToMachine := make(map[string]clusterv1.Machine)
+	for _, machine := range machineList.Items {
+		infraRef := machine.Spec.InfrastructureRef
+		infraMachineNameToMachine[infraRef.Name] = machine
+	}
+
+	for i := range infraMachines {
+		infraMachine := &infraMachines[i]
+		ownerRefs := infraMachine.GetOwnerReferences()
+		hasOwnerMachine := false
+		for _, ownerRef := range ownerRefs {
+			if ownerRef.Kind == "Machine" && ownerRef.APIVersion == clusterv1.GroupVersion.String() {
+				hasOwnerMachine = true
+				break
+			}
+		}
+
+		if !hasOwnerMachine {
+			machine, ok := infraMachineNameToMachine[infraMachine.GetName()]
+			if !ok {
+				return errors.Errorf("failed to patch ownerRef for infraMachine %q because no Machine has an infraRef pointing to it", infraMachine.GetName())
+			}
+			// Set the owner reference on the infraMachine to the Machine since the infraMachine is created and owned by the infraMachinePool.
+			infraMachine.SetOwnerReferences([]metav1.OwnerReference{
+				*metav1.NewControllerRef(&machine, machine.GroupVersionKind()),
+			})
+
+			patchHelper, err := patch.NewHelper(infraMachine, r.Client)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create patch helper for infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+			}
+
+			if err := patchHelper.Patch(ctx, infraMachine); err != nil {
+				return errors.Wrapf(err, "failed to patch infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+			}
+		}
+	}
+
+	return nil
+}
+
+// getNewMachine creates a new Machine object. The name of the newly created resource is going
+// to be created by the API server, we set the generateName field.
+func getNewMachine(mp *expv1.MachinePool, infraMachine *unstructured.Unstructured) *clusterv1.Machine {
+	infraRef := corev1.ObjectReference{}
+	if infraMachine != nil {
+		infraRef.APIVersion = infraMachine.GetAPIVersion()
+		infraRef.Kind = infraMachine.GetKind()
+		infraRef.Name = infraMachine.GetName()
+		infraRef.Namespace = infraMachine.GetNamespace()
+	}
+
+	annotations := mp.Spec.Template.Annotations
+	if annotations == nil {
+		annotations = make(map[string]string)
```

Comment:
not sure we need to initialize annotations here if we're not going to set any (any future function calls that add annotations should take care of initializing the map if it's nil)